automation/e2e-functests.setup.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash -xeu
+
+# This script prepares an ephemeral cluster with
+# latest changes CNAO and latest commit KubeVirt on top.
+# And clones the desired component repository
+#
+# This script also exports:
+# $KUBECONFIG
+#  kubeconfig full path so you can use`kubectl` binary or
+#  CNAO's `./cluster/kubectl.sh` script directly.
+#
+# $TMP_PROJECT_PATH
+# Latest changes CNAO temp directory
+#
+# $TMP_COMPONENT_PATH
+# The desired component temp directory
+#
+# Example:
+# export COMPONENT="KubeMacPool"
+# source automation/e2e-functest.setup.sh
+# cd ${TMP_COMPONENT_PATH}
+# KUBECONFIG=$KUBECONFIG make functests
+
+source hack/components/git-utils.sh
+source hack/components/yaml-utils.sh
+
+source automation/check-patch.setup.sh
+
+teardown() {
+    #make cluster-down
+    # rm -rf "$kubevirt_temp_dir"
+    # rm -rf "$component_temp_dir"
+    echo hi
+}
+
+trap teardown EXIT SIGINT SIGTERM SIGSTOP
+
+export COMPONENT=${COMPONENT:-kubemacpool}
+
+# Move to latest changes CNAO project temp dir
+cd ${TMP_PROJECT_PATH}
+
+# Spin up Kubernetes cluster
+export KUBEVIRT_PROVIDER='k8s-1.17'
+make cluster-down cluster-up
+
+# Export .kubeconfig full path, so it will be possible
+# to use 'kubectl' directly from the component directory path
+export KUBECONFIG=$(./cluster/kubeconfig.sh)
+
+#Deploy CNAO latest changes
+make cluster-operator-push
+make cluster-operator-install
+
+# Deploy KubeVirt latest stable release
+export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases | grep tag_name | grep -v -- - | sort -V | tail -1 | awk -F':' '{print $2}' | sed 's/,//' | xargs)
```

Comment:
Move this to deploy-kubevirt.sh

Reply:
I add `-u` flag for `deploy-kubevirt.sh` it is mandatory to export KUBEVIRT_VERSION,
and this is the version that will be deployed.
I did not put this in that script because I wanted to allow deploying specific version and in case we want the latest 
the use should figure which version is that and pass it.

How about adding a logic that will check if `KUBEVIRT_VERSION=latest-stable` then we use that `curl..` command?

Reply:
if i understood you right, you can use the method to get latest as you did, unless
KUBEVIRT_VERSION already has value given by the user

Reply:
Right